iceberg/iceberg-catalog/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -176,11 +180,10 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
 
     CommitStatus commitStatus = CommitStatus.FAILURE;
     boolean updateHiveTable = false;
-    HiveCommitLock commitLock = null;
 
+    HiveLock lock = lockObject(metadata);
```

Comment:
are we missing `Core: Avoid creating new metadata file on registerTable (#6591)`here? 
````
boolean newTable = base == null;
    String newMetadataLocation = writeNewMetadataIfRequired(newTable, metadata);
````

Reply:
Nice catch! I don't know how, but I have skipped it. Let me add to the list. It was part of the 1.2.1 release.

Reply:
I added it in commit https://github.com/apache/hive/pull/4291/commits/484c5e764ec40adf7a579260ec73e9952eb60e91

Thank you for finding it.

Reply:
Fixed.